README.md

```diff
@@ -7,12 +9,131 @@ I heard you liked IRC & Discord so I put some IRC in your Discord.
+
+### Optional pre-requisites
+
+- [nmap](https://nmap.org/)
+- [figlet](https://linux.die.net/man/6/figlet)
+- [Shodan](https://www.shodan.io/) API key
+- [IPInfo](https://ipinfo.io/) API key
+- Reverse proxying webserver
+  - [Caddy](https://caddyserver.com/) is highly recommended
+- [Docker](https://www.docker.com/) if deploying on Linux
+
+### Install
+
+1. Clone/download [the repository](https://github.com/edfletcher/drc)
+2. Run `npm install` in the project directory
+3. Create `config/local.json` and set the appropriate configuration
+   - See [`config/default.js`](https://github.com/edfletcher/drc/blob/main/config/default.js) for examples and commentary
+
+## Setup
+
+In the configuration, `botId` is the “Application ID” in the Discord Developer portal.
+
+The Discord bot must have “Message Content Intent” & “Privileged Gateway Intents” enabled to function correctly.
+
+### Bot channel
+
+1. Create a special channel in no category (or at least not in any of the special categories you’ll create in the next step) for the bot’s status, notice et. al messages.
+2. Add this channel’s ID to the configuration in the (truly terribly-named) `irc.quitMsgChanId` [field](https://github.com/edfletcher/drc/blob/8034fd6e9727953f85ce3fd5754df796f4b6bf7b/config/default.js#L73).
+
+### Server categories
+
+1. Add a category for each IRC server, named for that server e.g. `irc.libera.chat`.
+1. Add a channel in that category for each IRC channel you wish to join on connect.
+1. Add to `config/channelXForms-NODE_ENV.json` with required name transforms! You can also adjust these from the client at any time with the `!channelXforms` command.
+
+### Allowed speakers
+
+To control which Discord users can speak as your IRC user, create a Role in your guild and add users to it you wish to allow to speak and use DRC in your server. Set the ID of this role as `config.app.allowedSpeakersRoleId`.
+
+### Configuration
+
+DRC uses the `config` module and as such follows [these rules](https://github.com/node-config/node-config/wiki/Configuration-Files#file-load-order) as to which configuration file will be used.
+
+`default.js` both specifies all default values as well as illustrates the expected structure. As a hobby project, some of the names have been very poorly chosen. You have my apologies.
+
+#### Critical configuration values
+
+The following configuration parameters must be set by you, the user for the system to function correctly:
+
+- `app.allowedSpeakersRoleId` or `app.allowedSpeakers`
+- `discord.botId`
+- `discord.guildId`
+- `discord.token`
+- `irc.quitMsgChanId`
+
+### Using private message channels
+
+If you create a category with a name that matches [this logic](https://github.com/edfletcher/drc/blob/main/discord.js#L347)\* - such as “Private Messages”, “privmsgs”, or “PMs”) - DRC will automatically create a channel in this category for each private message received (a la traditional IRC client’s “query window” feature).
```

Comment:
```suggestion
If you create a category with a name that matches [this logic](https://github.com/edfletcher/drc/blob/d4d7e8811eeb70c0fd37edf94d006744db1e61a4/discord.js#L629)\* - such as “Private Messages”, “privmsgs”, or “PMs”) - DRC will automatically create a channel in this category for each private message received (a la traditional IRC client’s “query window” feature).
```

(Fixing a mistake of mine in the source documentation, not using a permalink means that the line reference moves over time and becomes incorrect. This fixes that.)